cromulent/extract.py

```diff
@@ -203,8 +208,11 @@ def normalize_dimension(dimensions):
 		if values:
 			used_systems += 1
 	if used_systems != 1:
-		warnings.warn('*** dimension used a mix of metric, imperial, and/or unknown: '\
-						'%r' % (dimensions,))
+		if source:
+			warnings.warn(f'*** dimension used a mix of metric, imperial, and/or unknown: {dimensions!r}; source is {source!r}')
```

Comment:
Please reformat to not use 3.x only `f'...'` construct

Reply:
Argh! Sorry. I keep forgetting crom is 2.x. Any thoughts on the impact of the impending deprecation?

Reply:
No worries :) It should work in 2 and 3 even after 2 is end of life, at least for a while, unless there's a significant improvement that isn't backwards compatible.